build/main.go

```diff
@@ -208,21 +212,65 @@ func javaBuild(ctx context.Context, client *dagger.Client, hostDirectory *dagger
 }
 
 func phpBuild(ctx context.Context, client *dagger.Client, hostDirectory *dagger.Directory) error {
-	container := client.Container().From("php:8-cli").
-		WithDirectory("/src", hostDirectory.Directory("flipt-php")).
-		WithEnvVariable("COMPOSER_ALLOW_SUPERUSER", "1").
-		WithExec([]string{"sh", "-c", "curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer"}).
-		WithWorkdir("/src").
-		WithExec([]string{"composer", "install"})
+	if tag == "" {
+		return fmt.Errorf("tag is not set")
+	}
+	// because of how Composer works, we need to create a new repo that contains
+	// only the php client code.
+	targetRepo := os.Getenv("TARGET_REPO")
+	if targetRepo == "" {
+		targetRepo = "https://github.com/flipt-io/flipt-php.git"
+	}
 
-	var err error
+	targetTag := strings.TrimPrefix(tag, "refs/tags/flipt-php-")
+
+	pat := os.Getenv("GITHUB_TOKEN")
+	if pat == "" {
+		return errors.New("GITHUB_TOKEN environment variable must be set")
+	}
+
+	encodedPAT := base64.URLEncoding.EncodeToString([]byte("pat:" + pat))
+
+	var gitUserName = os.Getenv("GIT_USER_NAME")
+	if gitUserName == "" {
+		gitUserName = "flipt-bot"
+	}
+
+	var gitUserEmail = os.Getenv("GIT_USER_EMAIL")
+	if gitUserEmail == "" {
+		gitUserEmail = "dev@flipt.io"
+	}
+
+	git := client.Container().From("golang:1.21.3-bookworm").
+		WithExec([]string{"git", "config", "--global", "user.email", gitUserEmail}).
+		WithExec([]string{"git", "config", "--global", "user.name", gitUserName}).
+		WithExec([]string{"git", "config", "--global",
+			"http.https://github.com/.extraheader",
+			fmt.Sprintf("AUTHORIZATION: Basic %s", encodedPAT)})
```

Comment:
I would get this in a Dagger secret:
https://docs.dagger.io/723462/use-secrets/

Otherwise, we're probably leaking this stuff in dagger output.